src/cdk/a11y/_a11y.scss

```diff
@@ -28,4 +28,10 @@
   @media (-ms-high-contrast: $target) {
     @content;
   }
+
+  @at-root {
+    .cdk-high-contrast-#{$target} #{&} {
```

Comment:
While we're at it, should we have some compile-time validation that the target is valid?

Reply:
This produces an unintended effect for cases where `#{&}` includes multiple comma-delimited selectors. 

For example, consider the following Sass:

```scss
.element-a, .element-b {
  @include cdk-high-contrast() {
    background: blue;
  }
}
```

This will actually produce the following:

```scss
.cdk-high-contrast .element-a, .element-b {
  background: blue;
}
```

In short, Sass is not prepending `.cdk-high-contrast` to each selector and instead just putting it once before the whole rule.

Reply:
Looks like this fix may be viable, where we loop over each item in the active selector:

```scss
  @at-root {
    @each $i in & {
      .cdk-high-contrast #{$i} {
        @content
      }
    }
  }
```

Reply:
Thanks for looking into the failures. Using `@each` here will duplicate the content for each part of the compound selector. Instead I did this:
```scss
  @at-root {
    $selector-context: #{&};
    .cdk-high-contrast-#{$target} {
      #{$selector-context} {
        @content
      }
    }
  }
```

For the "body not present yet" issue, I'm going to just add the check for now and we can figure out if further action is necessary as a follow-up. I suspect this is really just an issue with tests and not with real applications.